tests/marray/marray_constructor.h

```diff
@@ -31,81 +31,125 @@ template <typename DataT, typename NumElementsT>
 class run_marray_constructor_test {
   static constexpr std::size_t NumElements = NumElementsT::value;
 
-  using marray_t = sycl::marray<DataT, NumElements>;
+  static constexpr size_t num_test_cases = 7;
 
- private:
-  template <size_t num_elements = NumElements,
-            std::enable_if_t<num_elements == 1, bool> = true>
-  void check_constexpr_single_element() {}
+  static constexpr const char* check_names[num_test_cases] = {
+      "default constructor",
+      "scalar constructor",
+      "variadic constructor (NumElements DataT instances)",
+      "variadic constructor (one DataT instance, one marray instance)",
+      "variadic constructor (one marray instance, one DataT instance)",
+      "copy constructor",
+      "copy constructor rval reference"};
 
-  template <size_t num_elements = NumElements,
-            std::enable_if_t<num_elements != 1, bool> = true>
-  void check_constexpr_single_element() {
-    //  one DataT instance, one marray instance
-    {
-      constexpr sycl::marray<DataT, num_elements - 1> ma_const =
-          marray_common::iota_marray<DataT, num_elements - 1, 2>();
-      constexpr marray_t ma{1, ma_const};
-      marray_t ma_inc;
-      std::iota(ma_inc.begin(), ma_inc.end(), 1);
-      CHECK(value_operations::are_equal(ma_inc, ma));
-    }
-
-    // one marray instance, one DataT instance
-    {
-      constexpr sycl::marray<DataT, num_elements - 1> ma_const =
-          marray_common::iota_marray<DataT, num_elements - 1, 1>();
-      constexpr marray_t ma{ma_const, DataT(num_elements)};
-      marray_t ma_inc;
-      std::iota(ma_inc.begin(), ma_inc.end(), 1);
-      CHECK(value_operations::are_equal(ma_inc, ma));
-    }
-  }
-
- public:
-  void operator()(const std::string&) {
-    INFO("for number of elements \"" << NumElements << "\": ");
+  using marray_t = sycl::marray<DataT, NumElements>;
 
+  template <typename IteratorT>
+  static void run_checks(IteratorT results) {
     // default constructor
     {
       marray_t ma;
-      CHECK(value_operations::are_equal(ma, DataT{}));
+      *(results++) = value_operations::are_equal(ma, DataT{});
```

Comment:
`()` are redundant but it might be clearer too. So no problem.

Reply:
You are absolutely right, but since they are unary operators on each side of the expression I think it's one of those cases where the precedence rules get hard to remember, so I personally prefer the parenthesis. 😄